modules/local-notifications/src/main/java/com/gluonhq/attach/localnotifications/impl/IOSLocalNotificationsService.java

```diff
@@ -52,10 +72,42 @@ protected void unscheduleNotification(Notification notification) {
     
     @Override
     protected void scheduleNotification(Notification notification) {
+        createImageInAssets(notification.getImageInputStream());
         registerNotification(notification.getTitle() == null ? "" : notification.getTitle(), 
                 notification.getText(), notification.getId(), notification.getDateTime().toEpochSecond());
     }
-    
+
+    private void createImageInAssets(InputStream imageInputStream) {
+        File file = new File(assetsFolder, NOTIFICATION_IMAGE);
+        if (file.exists()) {
+            file.delete();
+        }
+
+        try (imageInputStream) {
+            copyFile(imageInputStream, file.getAbsolutePath());
+        } catch (IOException ex) {
+            LOGGER.log(Level.WARNING, "Error reading image file", ex);
+        }
+    }
+
+    private void copyFile(InputStream is, String pathEnd)  {
+        if (is == null) {
+            return;
+        }
+
+        try (OutputStream os = new FileOutputStream(pathEnd)) {
+            byte[] buffer = new byte[1024];
+            int length;
+            while ((length = is.read(buffer)) > 0) {
+                os.write(buffer, 0, length);
+            }
+            os.flush();
+        } catch (IOException ex) {
+            LOGGER.log(Level.WARNING, "Error copying file", ex);
```

Comment:
IOException is catched here, but also in the caller (line 88). If an IO exception occurs during e.g. is.read(buffer), the caller won't know.

Reply:
Same as above